Mathlib/MeasureTheory/MeasurableSpace/Basic.lean

```diff
@@ -865,6 +865,24 @@ theorem measurable_update_left {a : δ} [DecidableEq δ] {x : π a} :
     Measurable (update · a x) :=
   measurable_update'.comp measurable_prod_mk_right
 
+@[measurability, fun_prop]
```

Comment:
I'm not sure these lemmas are enough to let `fun_prop` work well. For instance,
```
import Mathlib
lemma foo {α β : Type*} {f : α → β} [MeasurableSpace α] [MeasurableSpace β] (hf : Measurable f) 
    (s : Set α) : Measurable (s.restrict f) :=
  -- by fun_prop -- fails
  hf.comp measurable_subtype_coe
```
Do you think this lemma (and its friends) would be worth adding?

Reply:
I didn't expect `fun_prop` to work here. I use restrictions as projections in a `Pi` space (the function `f ↦ s.restrict f`), and prove these projections are measurable. But this says nothing about the measurability of the restriction (`a ↦ s.restrict f a`). It seems nobody ever needed it, but I just remembered I used it in my first ever Lean project so I'll add them for completeness.

Reply:
But I don't think it really makes sense to add those in the preorder case.